pandas/core/algorithms.py

```diff
@@ -431,6 +432,13 @@ def isin(comps: AnyArrayLike, values: AnyArrayLike) -> np.ndarray:
         return cast("Categorical", comps).isin(values)
 
     comps, dtype = _ensure_data(comps)
+    if is_numeric_dtype(comps):
+        try:
+            # Try finding a dtype which would not change our values
+            values, _ = maybe_upcast(values, dtype=dtype)
+            dtype = values.dtype
+        except (ValueError, TypeError):
```

Comment:
what cases raise?

Reply:
``TypeError`` when values is an array with ``pd.NaT`` (``test_isin_nan_common_float64`` in ``pandas/tests/indexes/test_base.py``)
``ValueError`` if values contains strings, which can't be converted to int (``test_isin_level_kwarg`` in same file for example)

Reply:
not a big fan of this, can we not be explict here on the dtype checks (as we are elsewhere).?

Reply:
Could use ``maybe_convert_numeric``` but would have to keep a ``try except`` block too. Is there a function I am not aware of to check, if we can convert to numeric from object without try except?